src/irmin-pack/pack.ml

```diff
@@ -115,17 +115,18 @@ struct
   type index = Index.t
 
   type 'a t = {
-    block : IO.t;
+    mutable block : IO.t;
     index : Index.t;
     dict : Dict.t;
     lock : Lwt_mutex.t;
     mutable open_instances : int;
   }
 
   let clear t =
-    IO.clear t.block;
-    Index.clear t.index;
-    Dict.clear t.dict
+    if IO.offset t.block <> 0L then (
+      IO.clear t.block;
+      Index.clear t.index;
+      Dict.clear t.dict )
```

Comment:
Unrelated to this PR, but I think there's a race condition here: we read in the order `Dict` → `Index` → `Block` and clear in the order `Block` → `Index` → `Dict`, so I think it's possible to read from a dangling hash in Index during this process.

Reply:
> we read in the order Dict → Index → Block

I think we always read the index first.

I moved the `Index.clear` first to avoid the dangling hash issue you mentioned.

Reply:
> I think we always read the index first.

Indeed, sorry (I was mixing `Dict` with `Lru` here...).